packages/scaffolder/docs/3-expressions.md

```diff
@@ -0,0 +1,150 @@
+[helper package](https://github.com/helpers/handlebars-helpers) to make it
+easier to work with expressions.
+
+For the purposes of Alley Scaffolder, variables and other expressions are
+surrounded by `{{ }}`. This syntax is similar to GitHub Actions, but is not
+compatible with GitHub Actions expressions.
+
+You can see expressions in use in the [Feature Configuration](./2-features.md).
+Let's use an `if` conditional file as an example:
+
+```yaml
+files:
+  - source: test.stub
+    if: "{{ inputs.tests }}"
+    destination: tests/Features/{{ psrClassFilename inputs.caseStudyName prefix="" suffix="Test.php" }}
+```
+
+In this example, the `if` conditional is used to conditionally scaffold a file
+based on the user's input. If the user selects to scaffold tests, the file will
+be scaffolded. If not, the file will be skipped.
+
+## Helpers
+
+Helper are the most powerful part of expressions. Helpers are used to modify
+the user's input. For example, if you want to convert a user's input to a
+valid WordPress file name, you can use the `wpClassFilename` helper.
+
+### `wpClassFilename`
+
+Converts a string to a valid WordPress file name. By default, the file name will
+be prefixed with `class-` and suffixed with `.php` but can be overridden with
+the `prefix` and `suffix` arguments.
+
+```
+{{ wpClassFilename "User Input" }} => "class-user-input.php"
+{{ wpClassFilename "Folder/User Input" }} => "folder/class-user-input.php"
+{{ wpClassFilename "Folder/Subfolder/User Input" }} => "folder/subfolder/class-user-input.php"
+{{ wpClassFilename "User Input" prefix="prefix-" }} => "prefix-user-input.php"
+{{ wpClassFilename "User Input" prefix="prefix-" suffix=".suffix" }} => "prefix-user-input.suffix"
+```
+
+### `wpClassName`
+
+Converts a string to a valid WordPress class name. This filter will strip any
+preceding folder (which becomes a namespace) along with any invalid characters
+and converts the string to Pascal_Case.
+
+```
+{{ wpClassName "User Input" }} => "User_Input"
+{{ wpClassName "Folder/User Input" }} => "User_Input"
+{{ wpClassName "Folder/Subfolder/User Input" }} => "User_Input"
+```
+
+The filter also supports a `prefix` and `suffix` argument to add a prefix or
+suffix to the string.
+
+```
+{{ wpClassName "User Input" prefix="Prefix_" suffix="_Suffix" }} => "Prefix_User_Input_Suffix"
+```
+
+### `wpNamespace`
+
+Converts a string to a valid WordPress namespace. This filter will strip any
+invalid characters and convert the string to Pascal_Case. The last part of the input will be stripped and used as the class name.
+
+```
+{{ wpNamespace "User Input" }} => ""
+{{ wpNamespace "Folder/User Input" }} => "Folder"
+```
+
+The filter also supports a base namespace argument to add a prefix to the
+namespace.
+
+```
+{{ wpNamespace "User Input" prefix="Feature\" }} => "Feature\\"
+{{ wpNamespace "Folder/User Input" prefix="Feature\" }} => "Feature\\Folder"
+```
+
+### `psrClassFilename`
```

Comment:
imo this should be `psr4ClassFilename` - PSR-4 is a newer version of the PSR-0 autoloader standard, and there's always a chance they release a newer newer version in the future